### PR TITLE
[Package Signing] Add tests for repository signing

### PR DESCRIPTION
**Pull request by loic-sharma**
Here is an example run: https://nuget.visualstudio.com/NuGetBuild/_build/results?buildId=34801&view=logs

Depends on: https://nuget.visualstudio.com/_git/NuGetBuild/pullrequest/543?_a=overview

Fixes https://github.com/NuGet/Engineering/issues/1205
Fixes https://github.com/NuGet/Engineering/issues/1206

### CODE REVIEW

**Comment by loic-sharma** [comment]
🔔 Ready for review! I verified this on DEV.

File: src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj [hunk 1]
```diff
@@ -101,6 +101,9 @@
     <Content Include="Support\NuGetExe\nuget.4.6.1.exe">
```
Comment (chenriksson) [inline src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj]: Is `nuget.4.6.1.exe` still used elsewhere? Does it need to be?

Comment (loic-sharma) [reply src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj]: Good question. The pattern has been to keep the nuget client around, but I'm not sure why. I'll do some digging.

Comment (loic-sharma) [reply src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj]: I did a scan through, I couldn't find anything using the previous clients. I went ahead and remove all unnecessary clients.

File: src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj [hunk 2]
```diff
@@ -95,10 +95,7 @@
     <Content Include="Support\NuGetExe\nuget.4.1.0.exe">
       <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
     </Content>
-    <Content Include="Support\NuGetExe\nuget.4.3.0-preview1-4044.exe">
-      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
-    </Content>
-    <Content Include="Support\NuGetExe\nuget.4.6.1.exe">
+    <Content Include="Support\NuGetExe\nuget.4.7.0.exe">
```
Comment (scottbommarito) [inline src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj]: Out of curiosity, why are we keeping `4.1.0` but replacing the other versions?

Comment (loic-sharma) [reply src/NuGet.Services.EndToEnd/NuGet.Services.EndToEnd.csproj]: 4.1.0 is still used by tests due to a behavior change in the client.

**Comment by loic-sharma** [comment]
Here is an example run: https://nuget.visualstudio.com/NuGetBuild/_build/results?buildId=34801&view=logs

Depends on: https://nuget.visualstudio.com/_git/NuGetBuild/pullrequest/543?_a=overview

Fixes https://github.com/NuGet/Engineering/issues/1205
Fixes https://github.com/NuGet/Engineering/issues/1206

**Pull request merged**